contracts/src/Teleporter/TeleporterMessenger.sol

```diff
@@ -644,18 +647,20 @@ contract TeleporterMessenger is ITeleporterMessenger, ReentrancyGuards {
 
         // Store the fee asset and amount to be paid to the relayer of this message upon receiving the receipt.
         // Also store the message hash so that it can be retried until we get receipt of its delivery.
+        TeleporterFeeInfo memory adjustedFeeInfo = TeleporterFeeInfo({
+            contractAddress: feeInfo.contractAddress,
+            amount: adjustedFeeAmount
+        });
```

Comment:
Similar question here regarding gas differences when declaring the `TeleporterFeeInfo` as `memory` vs. constructing it inline in the `SentMessageInfo` construction.